Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/Doctrine/ConditionGenerator.php

```diff
@@ -130,7 +130,11 @@ protected function getNodeByIdentifier($nodeIdentifier)
         $this->securityContext->withoutAuthorizationChecks(function () use ($nodeIdentifier, $context, &$node) {
             $node = $context->getNodeByIdentifier($nodeIdentifier);
         });
-        $context->getFirstLevelNodeCache()->setByIdentifier($nodeIdentifier, null);
+        // we need to ensure that $node is never stored in the node cache
+        $context->getFirstLevelNodeCache()->removeNodeFromIdentifierCache($nodeIdentifier);
```

Comment:
@skurfuerst Couldn't this actually cause more queries under normal conditions now, because the nodes have to be queried again after the privilege check?

Reply:
Maybe we should do both removals inside the null condition.